docs/setup_local_development.md

```diff
@@ -0,0 +1,151 @@
+- [pytest](https://docs.pytest.org/en/6.2.x/) - for easy unit testing
+- [flake8](https://flake8.pycqa.org/en/latest/) - for code style enforcement
+
+All of these can be installed from the `requirements_dev.txt` file in `cloud-functions/` directory. Simply run `pip install -r requirements_dev.txt` from `cloud-functions/`, or `pip install -r ../requirements_dev.txt` from within a Cloud Function subdirectory.
+
+#### Running Cloud Functions Locally
+
+During development it can be helpful to run Cloud Functions on a local machine in order to test them without having to deploy to GCP. This can be done using the [Functions Framework](https://cloud.google.com/functions/docs/functions-framework). To run a Cloud Function locally simply navigate into its root directory, activate its virtual environemnt, and run `functions-framework --target <MY-FUNCTION-NAME> --debug`.
+
+#### Cloud Function Unit Testing
+
+As mentioned in [Cloud Function Directory Structure](#cloud-function-directory-structure) every Cloud Function has unit testing in a `test_<FUNCTION-NAME>.py` file. We use [pytest](https://docs.pytest.org) to run these unit tests. Pytest is included in the [Development Dependencies](#development-dependencies), but can also be installed with `pip install pytest`. To run the unit tests for a Cloud Function navigate to its root directory and simply run `pytest`. To run the unit tests for all Cloud Function in this repository navigate to `phdi-google-cloud/cloud-functions/` and run `pytest`. Please note that merging into the `main` branch of this repository is automatically blocked if all unit tests are not passing, see [Continuous Integration (CI)](#continuous-integration-ci) for details on this.
+
+#### Pushing to Github
+
+To get access to push to Github, ask to get maintainer access to the repo for your Github account.
+
+### Infrastructure as Code (IaC)
+
+IaC is the practice of writing machine-readable code for infrastructure configuration. It offers numerous benefits including, allowing infrastructure to be tracked in source control, and the ability to easily create multiple identical instances our infrastructure. For more information in general about IaC this [Wikipedia page](https://en.wikipedia.org/wiki/Infrastructure_as_code) may be a good starting place. In this repository the `phdi-google-cloud/terraform/` directory contains full coverage for all of our GCP infrastructure with HashiCorp [Terraform](https://www.terraform.io/). This directory has the following structure:
+
+```bash
+terraform/
+├── modules/
+│   ├── cloud-functions/
+│   │   ├── main.tf
+│   │   └── variables.tf
+│   ├── fhir/
+│   │   ├── main.tf
+│   │   └── variables.tf
+│   ├── network/
+│   │   └── main.tf
+│   └── storage/
+│       ├── main.tf
+│       └── outputs.tf
+└── vars/
+    └── skylight/
+        ├── backend.tf
+        ├── main.tf
+        ├── variables.tf
+        └── ~outputs.tf
+```
```

Comment:
Terraform directory has changed significantly. Maybe we should finish the intended refactor before updating this section though.